File: Aplicaciones en Java/Listas Personalizadas/app/src/main/java/com/example/listas_adaptador/MainActivity.java
package com.example.listas_adaptador;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ListView;
import android.widget.Toast;

import com.example.listas_adaptador.adaptadores.MyAdapter;
import com.example.listas_adaptador.entidades.Alumno;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        ListView listView = findViewById(R.id.list_view);

        ArrayList<Integer> notasJuan = new ArrayList<>();
        notasJuan.add(80);
        notasJuan.add(85);
        notasJuan.add(90);

        ArrayList<Alumno> listAlumnos = new ArrayList<>();
        listAlumnos.add(new Alumno("Juan", 20,notasJuan));
        listAlumnos.add(new Alumno("María", 22,notasJuan));
        listAlumnos.add(new Alumno("Pedro", 21,notasJuan));

        MyAdapter adapter = new MyAdapter(this, listAlumnos);
        listView.setAdapter(adapter);

        listView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
                Alumno alumno = (Alumno) parent.getItemAtPosition(position);
                Toast.makeText(MainActivity.this, "Clic en el alumno: " + alumno.getNombre(), Toast.LENGTH_SHORT).show();
            }
        });

    }
}

File: Aplicaciones en Kotlin/Menus_listas_nav/app/src/main/java/com/example/menus_listas_nav/MainActivity.kt
package com.example.menus_listas_nav

import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.ArrayAdapter
import android.widget.ListView
import android.widget.Toast
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import com.example.menus_listas_nav.data.obtenerListaDeLibros
import com.example.menus_listas_nav.databinding.ActivityMainBinding
import com.google.android.material.shape.MaterialShapeDrawable


class MainActivity : AppCompatActivity() {
    private lateinit var binding: ActivityMainBinding


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }


        // Usa el ArrayAdapter por defecto
        val adapter = ArrayAdapter(
            this,
            android.R.layout.simple_list_item_1, // Layout predefinido para elementos de la lista
            obtenerListaDeLibros()
        )

        binding.list.adapter = adapter



        Log.d("DEBUG", "Lista de libros: ${obtenerListaDeLibros()}")


        binding.appBarMain.toolbar.setNavigationOnClickListener {
            binding.main.open()
        }

        binding.appBarMain.toolbar.setOnMenuItemClickListener { menuItem ->
            when (menuItem.itemId) {
                R.id.settings -> {
                    Toast.makeText(this, "Configuración", Toast.LENGTH_SHORT).show()
                    true
                }
                R.id.account -> {
                    Toast.makeText(this, "Cuenta", Toast.LENGTH_SHORT).show()
                    true
                }
                R.id.more -> {
                    Toast.makeText(this, "Más", Toast.LENGTH_SHORT).show()
                    true
                }
                R.id.version -> {
                    Toast.makeText(this, "Versión", Toast.LENGTH_SHORT).show()
                    true
                }
                else -> false
            }
        }
        binding.navView.setNavigationItemSelectedListener { menuItem ->
            // Handle menu item selected
            menuItem.isChecked = true
            if(menuItem.itemId == R.id.search_item){
                Toast.makeText(this," Search Title", Toast.LENGTH_SHORT).show()
            }
            binding.main.close()
            true
        }

    }
}